packages/forms/resources/js/components/markdown-editor.js

```diff
@@ -199,6 +215,169 @@ export default function markdownEditorFormComponent({
                 )
             }
 
+            if (hasLanguageAssistance) {
+                let oldValue = null
+                const markers = []
+
+                const debounce = (func, timeout = 300) => {
+                    let timer
+
+                    return (...args) => {
+                        clearTimeout(timer)
+                        timer = setTimeout(() => {
+                            func.apply(this, args)
+                        }, timeout)
+                    }
+                }
+
+                const getAllIndices = (str, val) => {
+                    const indices = []
+                    let i = -1
+
+                    while ((i = str.indexOf(val, i + 1)) !== -1) {
+                        indices.push(i)
+                    }
+
+                    return indices
+                }
+
+                const checkLanguage = (cm) => {
+                    const newValue = cm.getValue()
+
+                    if (oldValue === newValue) {
+                        return
+                    }
+
+                    this.moderateSentences = 0
+                    this.complexSentences = 0
+                    this.passivePhrases = 0
+                    this.adverbPhrases = 0
+                    this.complexPhrases = 0
+                    this.qualifiedPhrases = 0
+
+                    while (markers.length > 0) {
+                        try {
+                            markers.pop().clear()
+                        } catch (e) {
+                            console.warn(e)
+                        }
+                    }
+
+                    oldValue = newValue
+
+                    const lines = cm.doc.children[0].lines
+
+                    for (let line of lines) {
+                        let sentenceOffset = 0
+
+                        const sentences = getSentenceDifficulty(line.text)
+                        const lineNumber = cm.getLineNumber(line)
+                        const lineElement =
+                            this.$refs.editor.parentNode.parentNode.querySelectorAll(
```

Comment:
Shameful, I know. CM doesn't provide a way to get the line elements, and this event handler isn't given them either. Can't use renderLine (which does get a line element) because marking in that generates an infinite loop.